mobile/src/main/java/com/gelakinetic/mtgfam/fragments/DecklistFragment.java

```diff
@@ -167,188 +233,171 @@ public boolean onItemLongClick(AdapterView<?> parent, View view, int position, l
+
+        /* Add it to the decklist, either as a new CompressedDecklistInfo, or to an existing one */
+        if (mCompressedDecklist.contains(decklistInfo)) {
+            boolean added = false;
+            final int firstIndex = mCompressedDecklist.indexOf(decklistInfo);
+            final DecklistHelpers.CompressedDecklistInfo firstCard =
+                    mCompressedDecklist.get(firstIndex);
+            for (int i = 0; i < firstCard.mInfo.size(); i++) {
+                CardHelpers.IndividualSetInfo firstIsi = firstCard.mInfo.get(i);
+                if (firstIsi.mSetCode.equals(card.setCode) && firstIsi.mIsFoil.equals(card.foil)) {
+                    firstIsi.mNumberOf++;
+                    added = true;
+                    break;
                 }
-            } else {
-                mCompressedDecklist.add(new CompressedDecklistInfo(card, isSideboard));
             }
+            if (!added) {
+                firstCard.add(card);
+            }
+        } else {
+            mCompressedDecklist.add(new DecklistHelpers.CompressedDecklistInfo(card, isSideboard));
+        }
 
-            /* Sort the decklist */
-            Collections.sort(mCompressedDecklist, mDecklistChain);
-
+        /* The headers shouldn't (and can't) be sorted */
+        clearHeaders();
 
-            /* Save the decklist */
-            DecklistHelpers.WriteCompressedDecklist(getActivity(), mCompressedDecklist);
+        /* Load the card's price */
+        if (mShowTotalDecklistPrice) {
+            loadPrice(card.mName, card.setCode, card.mNumber);
+        }
 
-            /* Clean up for the next add */
-            mNumberField.setText("1");
-            mNameField.setText("");
+        /* Sort the decklist */
+        Collections.sort(mCompressedDecklist, mDecklistChain);
 
-            /* Update the headers */
-            setHeaderValues();
+        /* Save the decklist */
+        DecklistHelpers.WriteCompressedDecklist(getActivity(), mCompressedDecklist);
 
-            /* Redraw the new decklist with the new card */
-            mDecklistAdapter.notifyDataSetChanged();
+        /* Clean up for the next add */
+        mNumberOfField.setText("1");
+        mNameField.setText("");
 
-        } catch (FamiliarDbException e) {
-            handleFamiliarDbException(false);
-        } catch (NumberFormatException e) {
-            /* eat it */
+        /* Uncheck the foil box if it isn't locked */
+        if (!mCheckboxFoilLocked) {
+            mCheckboxFoil.setChecked(false);
         }
+
+        /* Redraw the new decklist with the new card */
+        setHeaderValues();
+        mListAdapter.notifyDataSetChanged();
+
     }
 
     /**
-     * read and compress the wishlist
+     * Total the cards, and set it. Get the setting for price. And load up the decklist.
      */
     @Override
     public void onResume() {
+
         super.onResume();
+        final String totalCards = ((CardDataAdapter) mListAdapter).getTotalCards() + " ";
```

Comment:
Ha, I don't think Familiar is the kind of high performance computing where shaving nanoseconds is going to matter. I only brought it up because you fixed similar code in DeckCounterFragment.java. Does the padding matter here?